### PR TITLE
Feature/advanced student pages

### PR DESCRIPTION
**Pull request by simvadnbu**
This adds some extra features to the student pages.

**What's new?**
- You can now edit/post required steps (arrival, inside, leaving).
- You can post/patch/delete global remarks for a building.
- When a tour is finished, you can view a modal of the files/description/comments that you made.

**What still needs to be added?**
The websockets & when refreshing, set the right building.

@TiboStr @sevrijss I changed a view + added a check in the clean method of the `RemarkAtBuilding` models, can you take a look if this was done correctly & if this has an influence on the tests? 

**Edit:**
This now also adds an open mail client link to the communication page.

### CODE REVIEW

**Review by GashinRS — APPROVED** [review]

**Comment by TiboStr** [comment]
Are there any issues you close with this PR?

File: backend/base/models.py [hunk 1]
```diff
@@ -357,6 +357,10 @@ def clean(self):
         super().clean()
         if not self.timestamp:
             self.timestamp = datetime.now()
+        if self.type == "AA" or self.type == "BI" or type == "VE":
```
Comment (sevrijss) [inline backend/base/models.py:360]: I though multiple picture inside the building were allowed?

Comment (TiboStr) [inline backend/base/models.py:360]: Why not `self.type == "OP"` as well?

File: backend/base/models.py [hunk 2]
```diff
@@ -357,6 +357,10 @@ def clean(self):
...
+            remark_instances = RemarkAtBuilding.objects.filter(building=self.building, student_on_tour=self.student_on_tour, type=self.type)
+            if remark_instances.count() == 1:
+                raise ValidationError("There already exists a remark of this type from this student on tour at this building.")
```
Comment (TiboStr) [inline backend/base/models.py:363]: * `from django.utils.translation import gettext_lazy as _`
* Put the string inside `_()`

More info: [wiki on i18n](https://github.com/SELab-2/Dr-Trottoir-4/wiki/i18n)

Comment (TiboStr) [reply backend/base/models.py:363]: I'll look later for a translation. You don't have to do it right now as git can't merge these translation files very well.

File: backend/base/models.py (hunk 1 above)
Comment (simvadnbu) [reply backend/base/models.py:360]: Yeah multiple pictures are allowed but there may only be one remark for AA, BI & VE. Only one remarks for arrival, inside & leaving is allowed.

Comment (simvadnbu) [reply backend/base/models.py:360]: You can have multiple global remarks (=OP) for a building, but you arrive, go inside & leave only once at a building.

Comment (sevrijss) [reply backend/base/models.py:360]: My understanding was multiple picture inside, but I don't think this matters a lot. More pictures can always be uploaded under a normal comment.

Comment (simvadnbu) [reply backend/base/models.py:360]: Yeah mulitple pictures of a single remark is still possible. It's just the remark (string) that is only allowed once.

**Comment by simvadnbu** [comment]
> Are there any issues you close with this PR?

I will link them, i will create an issue for the implementation of the websockets.

**Review by TiboStr — APPROVED** [review]

**Review by sevrijss — APPROVED** [review]

**Pull request merged**